### PR TITLE
[doc] Add M0 topo description to testbed overview doc

### PR DESCRIPTION
**Pull request by yaqiangz**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Describe details of M0 type topology.

#### How did you do it?
This added document explaining M0 topology.

### CODE REVIEW

File: docs/testbed/README.testbed.Overview.md [hunk 1]
```diff
@@ -166,6 +166,20 @@ The T2 type topology has variations:
 
 * The supervisor card does not have DUT ports.
 
+### M0 type topology
+
+The M0 type topology is to simulate a SONiC DUT running as a M0 device. For this type of topology, a set of DUT ports are connected to VMs simulating upstream M1 neighbors. Another set of the ports are connected to a PTF docker simulating downstream servers. Rest of the ports are connected to VMs simulating downstream Mx neighbors.
```
Comment (Blueve) [inline docs/testbed/README.testbed.Overview.md:171]: ...running as a `Management ToR Router` ... simulating upstream M1`(Management Leaf Router)` neighbors ... simulating downstream Mx`(BMC Management Router)` neighbors.

File: docs/testbed/README.testbed.Overview.md [hunk 2]
```diff
@@ -166,6 +166,20 @@ The T2 type topology has variations:
...
+
+**The PTF docker also has injected ports connected to the open vSwitch bridges interconnecting VMs and DUT ports. The injected ports can be used for both injecting packets to DUT and sniffing packets from DUT. Details of the injected ports will be explained in later sections.**
+
+![](./img/testbed-m0.png)
+
+* The DUT has 52 ports.
+* Requires 6 VMs.
+* The first 46 ports are connected to PTF docker simulating servers.
+* The next 4 ports are connected to 4 VMs simulating M1 devices. The connection to each of the upstream M1 is configured as a port-channel with single link.
+* The last 2 ports are connected to another 2 VMs simulating downstream Mx neighbors. No port-channel is configured for the links between DUT and Mx neighbors.
```
Comment (Blueve) [inline docs/testbed/README.testbed.Overview.md:181]: Why no port-channel?

**Review by Blueve — CHANGES_REQUESTED** [review]

File: docs/testbed/README.testbed.Overview.md [hunk 3]
```diff
@@ -166,6 +166,20 @@ The T2 type topology has variations:
...
+The M0 type topology is to simulate a SONiC DUT running as a Management ToR Router device. For this type of topology, a set of DUT ports are connected to VMs simulating upstream M1 (Management Leaf Router) neighbors. Another set of the ports are connected to a PTF docker simulating downstream servers. Rest of the ports are connected to VMs simulating downstream Mx (BMC Management Router) neighbors.
+
+**The PTF docker also has injected ports connected to the open vSwitch bridges interconnecting VMs and DUT ports. The injected ports can be used for both injecting packets to DUT and sniffing packets from DUT. Details of the injected ports will be explained in later sections.**
+
+![](./img/testbed-m0.png)
+
+* The DUT has 52 ports.
+* Requires 6 VMs.
+* The first 46 ports are connected to PTF docker simulating servers.
+* The next 4 ports are connected to 4 VMs simulating M1 devices. The connection to each of the upstream M1 is configured as a port-channel with single link.
```
Comment (Blueve) [inline docs/testbed/README.testbed.Overview.md:180]: 1. next 2 ports -> Mx
2. last 4 ports -> M1

Comment (yaqiangz) [reply docs/testbed/README.testbed.Overview.md]: Fixed

**Review by Blueve — APPROVED** [review]

**Pull request merged**